src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -1987,30 +2018,34 @@ public Task WriteToServerAsync(DataTable table, DataRowState rowState, Cancellat
             try
             {
                 statistics = SqlStatistics.StartTimer(Statistics);
+                
                 ResetWriteToServerGlobalVariables();
                 _rowStateToSkip = ((rowState == 0) || (rowState == DataRowState.Deleted)) ? DataRowState.Deleted : ~rowState | DataRowState.Deleted;
                 _rowSource = table;
                 _dataTableSource = table;
                 _rowSourceType = ValueSourceType.DataTable;
                 _rowEnumerator = table.Rows.GetEnumerator();
                 _isAsyncBulkCopy = true;
-                resultTask = WriteRowSourceToServerAsync(table.Columns.Count, cancellationToken); // It returns Task since _isAsyncBulkCopy = true;
+                
+                // It returns Task since _isAsyncBulkCopy = true;
+                return WriteRowSourceToServerAsync(table.Columns.Count, cancellationToken);
             }
             finally
             {
                 SqlStatistics.StopTimer(statistics);
             }
-            return resultTask;
         }
 
         private Task WriteRowSourceToServerAsync(int columnCount, CancellationToken ctoken)
         {
+            #if NETFRAMEWORK
```

Comment:
This behaviour is a little odd - it's never been in netcore (even back to S.D.S's first checkin in 2015.) The logic behind it is sound though. I'd suggest enabling it in netcore, but have the method return `Task.FromCanceled(ctoken)` instead.

Reply:
What's even better is that's the only usage of ADP.CreatedTaskWithCancellation so I can just delete that, too